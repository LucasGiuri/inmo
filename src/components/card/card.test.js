import Card from './card';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent, screen } from '@testing-library/react';

it('Card component match snapshot', () => {
	const img = 'asd.jpg';
	const description = 'description';
	const price = 120;
	const title = 'title';
	const { container, getByTestId } = render(<Card tittle={title} price={price} img={img} description={description} />)
	expect(getByTestId('price')).toBeTruthy();
	expect(getByTestId('price')).toHaveTextContent(`${price}`);
	expect(getByTestId('description')).toBeTruthy();
	expect(getByTestId('description')).toHaveTextContent(description);
	expect(getByTestId('img')).toBeTruthy();
	expect(container).toMatchSnapshot();
})