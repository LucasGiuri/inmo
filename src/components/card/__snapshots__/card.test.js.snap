// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardMedia-root makeStyles-media-2"
        data-testid="img"
        style="background-image: url(asd.jpg);"
        title="title"
      />
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
          data-testid="price"
        >
          $
          120
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          data-testid="description"
        >
          description
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <img
        src="bedIcon.png"
      />
    </div>
  </div>
</div>
`;
